dask/tests/test_tokenize.py

```diff
@@ -50,7 +52,44 @@ def check_contextvars():
         _seen.get()
 
 
+_TPE = None
+USE_TPE = None
+
+
+@pytest.fixture(
+    autouse=True, params=[pytest.param(True, marks=[pytest.mark.slow]), False]
+)
+def use_tpe(request):
+    global USE_TPE
+    USE_TPE = request.param
+
+
+def _init_tpe() -> ThreadPoolExecutor:
+    pytest.importorskip("psutil")
+    global _TPE
+    if _TPE is not None:
+        return _TPE  # type: ignore[unreachable]
+    cores = psutil.cpu_count(logical=False)
+    _TPE = ThreadPoolExecutor(cores * 4)
+    return _TPE
```

Comment:
This is a bit of a hack to avoid having to pass this through every function... hope nobody minds 🙈